### PR TITLE
Add Custom Line List Import API Docs

### PR DESCRIPTION
**Pull request by duytnguyendtn**
### Description

This pull request adds documentation on how to import custom line lists using an Astropy QTable

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: docs/notebook/import_data.rst [hunk 1]
```diff
@@ -48,3 +48,19 @@ For Imviz::
     imviz = Imviz()
     imviz.app
     imviz.load_data("/path/to/data/image.fits")
+
+
+Importing Custom Line Lists
+---------------------------
+Jdaviz comes with curated line lists built from the scientific community. If you can't find the lines you need, you can add your own!::
```
Comment (pllim) [inline docs/notebook/import_data.rst:55]:
```suggestion
===========================

Jdaviz comes with curated line lists built by the scientific community. If you cannot find the lines you need, you can add your own, as follows::
```

File: docs/notebook/import_data.rst [hunk 2]
```diff
@@ -48,3 +48,19 @@ For Imviz::
...
+
+    from astropy.table import QTable
+    import astropy.units as u
```
Comment (pllim) [inline docs/notebook/import_data.rst:58]: Nitpick.
```suggestion
    from astropy import units as u
```

File: docs/notebook/import_data.rst [hunk 3]
```diff
@@ -48,3 +48,19 @@ For Imviz::
...
+
+    lt = QTable()
```
Comment (pllim) [inline docs/notebook/import_data.rst:60]: `lt` is too cryptic. I propose a more legible variable name.
```suggestion
    my_line_list = QTable()
```

File: docs/notebook/import_data.rst [hunk 4]
```diff
@@ -48,3 +48,19 @@ For Imviz::
...
+    lt['linename'] = ['Hbeta','Halpha']
+    lt['rest'] = [4851.3, 6563]*u.AA
+    lt['redshift'] = u.Quantity(0.046) # Optional
+
+    viz.load_line_list(lt)
+    viz.spectral_lines
```
Comment (pllim) [inline docs/notebook/import_data.rst:66]:
```suggestion
    viz.spectral_lines  # Briefly explain what this does in the GUI here?
```

File: docs/notebook/import_data.rst [hunk 5]
```diff
@@ -48,3 +48,19 @@ For Imviz::
...
+
+Importing Custom Line Lists
+---------------------------
+Jdaviz comes with curated line lists built from the scientific community. If you can't find the lines you need, you can add your own!::
+
+    from astropy.table import QTable
+    import astropy.units as u
+
+    lt = QTable()
+    lt['linename'] = ['Hbeta','Halpha']
```
Comment (pllim) [inline docs/notebook/import_data.rst:61]: This is not the only way to construct a table. Maybe link to https://docs.astropy.org/en/latest/table/construct_table.html using intersphinx?

If you foresee people building a huge line list table, also link to https://docs.astropy.org/en/latest/table/index.html#performance-tips (if not, you don't have to). I'm talking about like maybe several thousand rows.

Comment (duytnguyendtn) [reply docs/notebook/import_data.rst:61]: Always a fan of additional documentation! I'll add those links, thanks!

File: docs/notebook/import_data.rst (hunk 3 above)
Comment (duytnguyendtn) [reply docs/notebook/import_data.rst:60]: Good point! I did make it clearer in https://github.com/spacetelescope/aas239-jwebbinar/pull/12, so probably a good idea here too

File: docs/notebook/import_data.rst (hunk 5 above)
Comment (duytnguyendtn) [reply docs/notebook/import_data.rst]: @pllim I never used intersphinx before (and it looks like there are a few places where links are still hardcoded). Any help to try to link to those pages would be much appreciated! `#newbie` :P

Comment (duytnguyendtn) [reply docs/notebook/import_data.rst]: Awesome! Much thanks! I got the Construct Table working, but I'm running into issues with the performance link. I found the `.. _astropy-table-performance:` header at the top of https://raw.githubusercontent.com/astropy/astropy/main/docs/table/performance.inc.rst, so I tried linking it using 
```
:ref:`<astropy:astropy-table-performance>`
```
but that didn't generate a link, and just generated raw text as "<astropy:astropy-table-performance>". Any ideas where I went wrong?

Comment (rosteen) [reply docs/notebook/import_data.rst]: I think there's essentially a 0% chance anyone will be adding so many lines that they'll need the table performance link, if it's not working then just remove it IMO.

Comment (duytnguyendtn) [reply docs/notebook/import_data.rst]: Works for me :)

File: docs/notebook/import_data.rst [hunk 6]
```diff
@@ -48,3 +48,21 @@ For Imviz::
     imviz = Imviz()
     imviz.app
     imviz.load_data("/path/to/data/image.fits")
+
+
+Importing Custom Line Lists
+===========================
+
+Jdaviz comes with curated line lists built by the scientific community. If you cannot find the lines you need, you can add your own by constructing an :ref:`astropy table <astropy:construct_table>`! For example::
```
Comment (pllim) [inline docs/notebook/import_data.rst:56]: Nitpick: It is easier for future review if you break this into mutiple lines. Also, I don't think the exclamation point is necessary.

```suggestion
Jdaviz comes with curated line lists built by the scientific community.
If you cannot find the lines you need, you can add your own by constructing
an :ref:`astropy table <astropy:construct_table>`; For example::
```

**Review by pllim — APPROVED** [review]
One remaining nitpick, but the content is fine, so approving. Thanks!

**Comment by pllim** [comment]
Thanks. LGTM! I'll merge as there is no need to run CI for doc only changes that does not involve doctest.

**Pull request merged**